### DISCUSSION TITLE
Access TypeScript types

### DISCUSSION THREAD

**Discussion by octava25** [sindresorhus/got#1550] [Q&A] [answered]
#### What would you like to discuss?

I have the following code in typescript:

```ts
const client = got.extend({
    ..., 
    handlers: [someHandler],
});
```

and handler function:

```ts
async function someHandler (options: ???, next: ???): Promise<???> {}
```

How can i access types of `HandlerFunction` or `HandlerFunction` type itself ? 
I would like to avoid `any` type or to ommit types at all.

#### Checklist

- [ x ] I have read the documentation.

**Comment by Giotino** [comment, marked as answer]
`HandlerFunction` type is exported by Got, you can use it like follows:

```ts
import { HandlerFunction } from 'got';
```

**Comment by Giotino** [comment]
If you were referring to the handler parameters types you can take a look at the definition of `HandlerFunction`

```ts
export declare type HandlerFunction = <T extends GotReturn>(options: NormalizedOptions, next: (options: NormalizedOptions) => T) => T | Promise<T>;
```

**Comment by octava25** [comment]
Thanks @Giotino. It seems that my typescript knowledge should be improved.

I tried to access it like this:

```ts
import got from 'got';
got.SomeType
```
